credentials/generate-revocation-set.py

```diff
@@ -198,27 +205,98 @@ def fetch_crl_from_url(url: str, timeout: int) -> x509.CertificateRevocationList
         logging.error('Failed to fetch a valid CRL')
 
 
-class DCLDClient:
+class DCLDClientInterface:
     '''
-    A client for interacting with DCLD using either the REST API or command line interface (CLI).
+    An interface for interacting with DCLD.
 
     '''
 
-    def __init__(self, use_rest: bool, dcld_exe: str, production: bool, rest_node_url: str):
+    def get_revocation_points(self) -> list[dict]:
+        '''
+        Get revocation points from DCL
+
+        Returns
+        -------
+        list[dict]
+            List of revocation points
+        '''
+        raise NotImplementedError
+
+    def get_revocations_points_by_skid(self, issuer_subject_key_id) -> list[dict]:
+        '''
+        Get revocation points by subject key ID
+
+        Parameters
+        ----------
+        issuer_subject_key_id: str
+            Subject key ID
+
+        Returns
+        -------
+        list[dict]
+            List of revocation points
+        '''
+        raise NotImplementedError
+
+    def get_certificate(self, issuer_name_b64: str, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
+        '''
+        Get certificate from DCL
+        '''
+        raise NotImplementedError
+
+    def get_first_approved_certificate(self, response: dict, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
```

Comment:
Under which circumstances would we get more than one certificate? The reason why I am asking is that if we have more than one certificate, is what we really want to get the first one (likely randomly ordered) from the set or do we want to pick the best one.

Reply:
I'll update to be get_only_approved_certificate, it will throw an error when no cert found or if >1 cert found.

Reply:
SG. Thank you.